### PR TITLE
feat(scripts/gdb): add draw unit debug info support

### PR DESCRIPTION
**Pull request by FASTSHIFT**
1. Add name field to `lv_draw_unit_t`
2. Add add drawing unit structure parsing.

```bash
-exec info draw_unit
Draw Unit: 0x614000000040, Name: VG_LITE
{
  base_unit = {
    next = 0x607000000100,
    target_layer = 0x60d00000c4e0,
    clip_area = 0x60d0000e471c,
    name = 0x555556b655e0 "VG_LITE",
    dispatch_cb = 0x5555564abca6 <draw_dispatch>,
    evaluate_cb = 0x5555564ac250 <draw_evaluate>,
    wait_for_finish_cb = 0x0,
    delete_cb = 0x5555564ac656 <draw_delete>
  },
  task_act = 0x60d0000e46e0,
  image_dsc_pending = 0x6040000000d0,
  grad_cache = 0x60c0000001c0,
  grad_pending = 0x604000000110,
  stroke_cache = 0x60c000000280,
  flush_count = 0,
  letter_count = 0,
  target_buffer = {
    width = 480,
    height = 480,
    stride = 1920,
    tiled = VG_LITE_LINEAR,
    format = VG_LITE_BGRX8888,
    handle = 0x0,
    memory = 0x7fffde033800,
    address = 3724752896,
    yuv = {
      swizzle = VG_LITE_SWIZZLE_UV,
      yuv2rgb = VG_LITE_YUV601,
      uv_planar = 0,
      v_planar = 0,
      alpha_planar = 0,
      uv_stride = 0,
      v_stride = 0,
      alpha_stride = 0,
      uv_height = 0,
      v_height = 0,
      uv_memory = 0x0,
      v_memory = 0x0,
      uv_handle = 0x0,
      v_handle = 0x0
    },
    image_mode = VG_LITE_NORMAL_IMAGE_MODE,
    transparency_mode = VG_LITE_IMAGE_OPAQUE,
    fc_buffer = {{
        width = 0,
        height = 0,
        stride = 0,
        handle = 0x0,
        memory = 0x0,
        address = 0,
        color = 0
      }, {
        width = 0,
        height = 0,
        stride = 0,
        handle = 0x0,
        memory = 0x0,
        address = 0,
        color = 0
      }, {
        width = 0,
        height = 0,
        stride = 0,
        handle = 0x0,
        memory = 0x0,
        address = 0,
        color = 0
      }},
    compress_mode = VG_LITE_DEC_DISABLE,
    index_endian = VG_LITE_INDEX_LITTLE_ENDIAN,
    paintType = VG_LITE_PAINT_ZERO,
    fc_enable = 0 '\000',
    scissor_layer = 0 '\000',
    premultiplied = 0 '\000'
  },
  global_matrix = {
    m = {{1, 0, 0}, {0, 1, 0}, {0, 0, 1}},
    scaleX = 5.73831721e-42,
    scaleY = -nan(0x7fc2c0),
    angle = 4.59163468e-41
  },
  global_path = 0x60f000000040,
  path_in_use = true
}
Draw Unit: 0x607000000100, Name: SW
{
  base_unit = {
    next = 0x0,
    target_layer = 0x0,
    clip_area = 0x0,
    name = 0x555556b545c0 "SW",
    dispatch_cb = 0x55555645d779 <dispatch>,
    evaluate_cb = 0x55555645d093 <evaluate>,
    wait_for_finish_cb = 0x0,
    delete_cb = 0x0
  },
  task_act = 0x0,
  idx = 0
}
```

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Review by kisvegabor — DISMISSED** [review]
Nice!

File: scripts/gdb/lvgl.py [hunk 1]
```diff
@@ -306,7 +309,65 @@ def invoke(self, args, from_tty):
...
+
+        # Scan available draw unit types
+        for index, type_name in type_names.items():
+            try:
+                types[index] = gdb.lookup_type(type_name)
+            except gdb.error:
+                pass
+
+        # Print the format string of the draw unit, if not found, use the default type
+        type = types.get(name, gdb.lookup_type("lv_draw_unit_t"))
```
Comment (XuNeo) [inline scripts/gdb/lvgl.py]:
```suggestion
        def lookup_type(name):
            try:
                return gdb.lookup_type(name)
            except gdb.error:
                return None

        types = {
            "DMA2D": lookup_type("lv_draw_dma2d_unit_t"),
            "NEMA_GFX": lookup_type("lv_draw_nema_gfx_unit_t"),
            "NXP_PXP": lookup_type("lv_draw_pxp_unit_t"),
            "NXP_VGLITE": lookup_type("lv_draw_vglite_unit_t"),
            "OPENGLES": lookup_type("lv_draw_opengles_unit_t"),
            "DAVE2D": lookup_type("lv_draw_dave2d_unit_t"),
            "SDL": lookup_type("lv_draw_sdl_unit_t"),
            "SW": lookup_type("lv_draw_sw_unit_t"),
            "VG_LITE": lookup_type("lv_draw_vg_lite_unit_t"),
        }

        type = types.get(name) or lookup_type("lv_draw_unit_t")
```

File: scripts/gdb/lvgl.py [hunk 2]
```diff
@@ -188,6 +188,9 @@ def screen_active(self):
         disp = self.lv_global["disp_default"]
         return disp["act_scr"] if disp else None
 
+    def draw_info(self):
```
Comment (XuNeo) [inline scripts/gdb/lvgl.py]: Let's add a drawunits iterator here:
```
def draw_units(self):
    unit = elf.lv_global["draw_info"]["unit_head"]

    # Iterate through all draw units
    while unit:
        yield unit
        unit = unit["next"]
```

File: scripts/gdb/lvgl.py [hunk 3]
```diff
@@ -306,7 +309,65 @@ def invoke(self, args, from_tty):
...
+            if g_lvgl_instance is None:
+                print("LVGL instance is not set.")
+                return
+
+            unit = g_lvgl_instance.draw_info()["unit_head"]
+
+            # Iterate through all draw units
+            while unit:
+                self.dump_draw_unit(unit)
+                unit = unit["next"]
```
Comment (XuNeo) [inline scripts/gdb/lvgl.py]:
```
for unit in g_lvgl_instance.draw_units():
    self.dump_draw_unit(unit)
```

File: scripts/gdb/lvgl.py [hunk 4]
```diff
@@ -306,7 +313,49 @@ def invoke(self, args, from_tty):
...
+
+    def __init__(self):
+        super(InfoDrawUnit, self).__init__(
+            "info draw_unit", gdb.COMMAND_USER, gdb.COMPLETE_EXPRESSION
+        )
+
+    def dump_draw_unit(self, draw_unit:gdb.Value):
+        # Dereference to get the string content of the name from draw_unit
+        name_address = draw_unit["name"]
+        name = name_address.string()
```
Comment (XuNeo) [inline scripts/gdb/lvgl.py]:
```suggestion
        name = draw_unit["name"].string()
```

**Review by XuNeo — APPROVED** [review]

**Review by kisvegabor — APPROVED** [review]

**Pull request merged**